parsing/lexer.mll

```diff
@@ -349,6 +352,8 @@ rule token = parse
           NATIVEINT (cvt_nativeint_literal (Lexing.lexeme lexbuf))
         with Failure _ ->
           raise (Error(Literal_overflow "nativeint", Location.curr lexbuf)) }
+  | (float_literal | int_literal) identchar*
```

Comment:
The star should probably be a plus.
The error `Invalid literal 0o18` seems weird when trying to lex `0o1823`